File: Data Structures & Algorithms/longest-consecutive-sequence/submission-8.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        hashset = set(nums)
        for entry in nums:
            if (entry - 1) not in hashset:
                length = 1
                while entry + length in hashset:
                    length += 1
                res = max(length, res)
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        int n = nums.size();
        vector<vector<int>> buckets(n + 1);
        unordered_map<int, int> freq;
        for (int& num : nums) {
            freq[num] += 1;
        }
        for (const auto& [key, val] : freq) {
            buckets[val].push_back(key);
        }
        vector<int> res;
        for (int i = n; i >= 0; i--) {
            for (int num : buckets[i]) {
                res.push_back(num);
                if (res.size() == k) {
                    return res;
                }
            }
        }
        return res;
    }
};
